lib/rule-context.js

```diff
@@ -98,7 +99,7 @@ function RuleContext(ruleId, eslint, severity, options, settings, ecmaFeatures)
      * The read-only ecmaFeatures shared across all rules
      */
     Object.defineProperty(this, "ecmaFeatures", {
-        value: Object.create(ecmaFeatures)
+        value: assign({}, ecmaFeatures)
```

Comment:
Prototype properties were causing problems based on how we pass around `ecmaFeatures`